aten/src/ATen/NumericUtils.h

```diff
@@ -80,6 +82,22 @@ inline C10_HOST_DEVICE bool _isnan(T val) {
   return val.isnan();
 }
 
+template <
+    typename T,
+    typename std::enable_if<std::is_same<T, at::Float8_e5m2fnuz>::value, int>::
+        type = 0>
+inline C10_HOST_DEVICE bool _isnan(T val) {
+  return val.isnan();
+}
+
+template <
+    typename T,
+    typename std::enable_if<std::is_same<T, at::Float8_e4m3fnuz>::value, int>::
+        type = 0>
+inline C10_HOST_DEVICE bool _isnan(T val) {
```

Comment:
```suggestion
inline C10_HOST_DEVICE bool _isnan(at::Float8_e4m3fnuz val) {
```

Reply:
Is there a reason why you suggest this change for the fnuz-suffixed type and no the other f8 types on lines 69-83?

Reply:
I think every overload that does `is_same<T, concrete_type>` could be update, it's just this that these ones are new in this PR.

Reply:
Could that perhaps be a follow-up PR that addresses your concern for all types that do this?